route/route.go

```diff
@@ -61,6 +64,31 @@ func CanConnect(hostport string) bool {
 	return true
 }
 
+// MightAuthenticate tests if a connection to host:port can initiate an handshake.
+func MightAuthenticate(hostport string, user string) bool {
+	ssh_config := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{
+			ssh.Password("ThisIsNotIntendedToBeAValidPasswordButWeDontReallyCare"),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	log.Debug("Dialing %s with user %s", hostport, user)
+	client, err := ssh.Dial("tcp", hostport, ssh_config)
+	if err != nil {
+		re_auth_failure := regexp.MustCompile(`handshake failed: ssh: unable to authenticate`)
+		if re_auth_failure.MatchString(err.Error()) {
```

Comment:
It would be better to merge the two lines i.e. `regexp.MustCompile(...).MatchString(...)`.